### ISSUE TITLE
[REQUEST]: Bulk Export API access for Pacstall

### ISSUE THREAD

**Issue by saenai255** [repology/repology-updater#1342]
# Title: Bulk Export API access for Pacstall

### Background:
Pacstall is a package manager designed for Debian and Ubuntu Linux distributions. It utilizes package specification files similar to Arch's PKGBUILD. To enhance the user experience, we currently use the Repology API to check whether the software package versions in our registry are up-to-date. However, as our package repository has grown, the number of API requests required for regular updates has increased significantly. This has led to throttling of our requests, affecting the performance and accuracy of our package management system.

### Request:
We kindly request access to the Repology Bulk Export API to streamline the process of updating package version information for Pacstall. We believe that using the Bulk Export API will significantly reduce the number of individual requests we need to make, thus avoiding throttling issues and improving the overall efficiency of our package registry.

### Intended Use Case:
We intend to use the Repology Bulk Export API to obtain comprehensive package version data for the software packages in our repository. By periodically fetching this data in bulk, we can efficiently update our package registry without overwhelming the API servers with frequent individual requests.

### Instructions Needed:
As the API documentation available on the Repology website provides limited details on the usage of the Bulk Export API, we kindly request the following information:

1. **API Endpoint:** Please provide us with the API endpoint URL for accessing the Bulk Export API.
2. **Authentication Requirements:** Let us know if the API requires any specific authentication method, such as API key, OAuth tokens, or other credentials.
3. **Rate Limiting:** Inform us about any rate limits or restrictions that apply to the Bulk Export API to help us optimize our data retrieval strategy.
4. **Data Format:** Please specify the format in which the bulk data is delivered (e.g., JSON, CSV, XML) and the structure of the response to facilitate seamless integration with our existing data parsing processes.
5. **Documentation or Code Examples:** If available, we kindly request additional documentation or code examples illustrating how to construct requests and process responses from the Bulk Export API.

### Importance of the Request:
Obtaining access to the Bulk Export API is crucial for enhancing the performance and reliability of Pacstall package manager. With access to the API and proper instructions, we can continue to provide our users with timely and accurate package version information, ensuring a smooth package management experience for Debian and Ubuntu users.

We appreciate your prompt attention to this request and look forward to receiving access to the Repology Bulk Export API along with the necessary instructions to integrate it seamlessly into our system.

**Note:**
If you need to contact us for any further information or have any questions regarding our request, please feel free to reach out to us via email at pacstall@pm.me. Additionally, you can join our [official Discord server](https://discord.com/invite/sWB6YtKyvW).

Thank you for your support.

**Comment by D-Brox** [comment]
This would only work with the official server. Personal servers would still need to use the old method.

This could be done tho, but it's not a priority at the moment.

**Comment by saenai255** [comment]
@D-Brox this is not true. We can offer our own API for personal servers that serves the cached bulk export.

**Comment by AMDmi3** [comment]
Doable (via https://github.com/repology/repology-exports), no ETA though. Please give me a hint on what export format would be most convenient for this task.

**Comment by saenai255** [comment]
`JSON` would be the easiest format for us to implement. Regarding [Repology Exports](https://github.com/repology/repology-exports), it seems to only parse a data file and doesn't actually download or connect to any API.

**Comment by AMDmi3** [comment]
JSON indeed, I'd like to know which fields you need there.

> Regarding [Repology Exports](https://github.com/repology/repology-exports), it seems to only parse a data file and doesn't actually download or connect to any API.

That's not intended to be run by you, it's intended to be setup on the Repology side to produce static json exports on a regular basis.

**Comment by saenai255** [comment]
Ideally, the more data we have, the better.

However, the essential fields are: `repo`, `subrepo`, `srcname`, `binname`, `visiblename`, `version`, `origversion`, `status`, and `projectname`. Here, `projectname` would be the path variable used when searching for a particular package. As an example, in the following endpoint `/api/v1/project/firefox` the `projectname` would be `firefox`

Our users rely on these fields to filter packages and find the correct one.

**Comment by saenai255** [comment]
Would it be fine if we used [this export script](https://github.com/saenai255/repology-export) until you finish yours? It fetches all the Repology project pages sequentially and aggregates them in an SQLite database. We plan to run this script about twice a day.

**Comment by AMDmi3** [comment]
@saenai255 yes, as long as you do no more than 1 rps and set an unique user-agent header

**Comment by Elsie19** [comment]
On our server, we're getting the error:
```go
2023/08/19 10:35:53 ERROR: Failed to export Repology projects: failed to fetch repology project page
Get "https://repology.org/api/v1/projects/": dial tcp 185.185.68.145:443: i/o timeout
```

but not on any of my personal computers, so I believe that our server is being blocked for some reason.

**Comment by AMDmi3** [comment]
> so I believe that our server is being blocked for some reason

These are listed one message above. If these are fixed, I need your IP to unblock.

**Comment by Elsie19** [comment]
Our IP is `104.238.165.50`.

**Comment by AMDmi3** [comment]
> Our IP is `104.238.165.50`.

Unblocked, please make sure you follow [terms of use](https://repology.org/api/v1).
